includes/class-wc-stripe-blocks-support.php

```diff
@@ -356,6 +411,11 @@ public function get_supported_features() {
 			$gateway = $gateways['stripe'];
 			return array_filter( $gateway->supports, [ $gateway, 'supports' ] );
 		}
+
+		if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() && empty( $gateways ) ) {
+			$gateway = new WC_Stripe_UPE_Payment_Gateway();
+			return array_filter( $gateway->supports, [ $gateway, 'supports' ] );
+		}
```

Comment:
This seems extremely backwards, but without this the supports array is empty when using UPE-based checkout. That would be a problem when we add support for Subscriptions and Pre-Orders when using UPE.

Ideally this should be removed, and I'm hoping the UPE gateway will become part of the `$gateways` array once #1856 is merged.

Reply:
Can't reproduce this behavior, debugged this for a while, and the UPE Gateway is also called `stripe` so it's always set and returns in the first if.

We can however change it to not access the property `payment_gateways ` directly:
```
$gateways = WC()->payment_gateways()->get_available_payment_gateways();
```

Reply:
Yep, I can no longer reproduce this after #1855 was merged, so I've removed this code 🙂